### PR TITLE
Gather and merge sign requests from repo builds

### PR DESCRIPTION
**Pull request by natemcmaster**
Part of https://github.com/aspnet/Coherence-Signed/issues/784

Depends on https://github.com/aspnet/BuildTools/pull/688

### CODE REVIEW

File: build/RepositoryBuild.targets [hunk 1]
```diff
@@ -23,13 +24,22 @@
   </Target>
 
   <Target Name="_BuildRepositories" DependsOnTargets="GetRepoBatches" Condition=" @(RepositoryBuildOrder->Count()) != 0 ">
+    <RemoveDir Directories="$(_SignRequestIntermediateOutputPath)" />
```
Comment (ryanbrandenburg) [inline build/RepositoryBuild.targets]: I assume you're just ensuring clean directories here?

File: build/RepositoryBuild.targets [hunk 2]
```diff
@@ -23,13 +24,22 @@
...
       BuildInParallel="true"
       StopOnFirstFailure="true"
       Targets="_BuildRepository"
       Properties="BuildGroup=%(BatchedRepository.BuildGroup);BuildNumber=$(BuildNumber);IsFinalBuild=$(IsFinalBuild);Configuration=$(Configuration)" />
 
+    <ItemGroup>
+      <SignRequests Include="$(_SignRequestIntermediateOutputPath)*.xml" />
+    </ItemGroup>
+
+    <MergeXmlFiles Files="@(SignRequests)" OutputPath="$(ArtifactsDir)signrequest.xml" />
```
Comment (ryanbrandenburg) [inline build/RepositoryBuild.targets]: Neat! Is doing this faster than sending multiple sign requests?

File: build/RepositoryBuild.targets (hunk 1 above)
Comment (natemcmaster) [reply build/RepositoryBuild.targets]: yes.

File: build/RepositoryBuild.targets (hunk 2 above)
Comment (natemcmaster) [reply build/RepositoryBuild.targets]: Yes. There is a fixed cost each time you have to round-trip to code-signing which is pretty high. So the more files you can send per batch, the better.

**Pull request merged**